static/js/services/count-unread-records.js

```diff
@@ -0,0 +1,46 @@
+angular.module('inboxServices').factory('CountUnreadRecords', function(
+  $q,
+  DB
+) {
+
+  'use strict';
+  'ngInject';
+
+  var TYPES = [ 'report', 'message' ];
+
+  var getTotals = function() {
+    return DB()
+      .query('medic-client/data_records_by_type', { group: true })
+      .then(function(response) {
+        var result = {};
+        response.rows.forEach(function(row) {
+          result[row.key] = row.value;
+        });
+        return result;
+      });
+  };
+
+  var getRead = function(type) {
+    var key = 'read:' + type + ':';
+    return DB({ meta: true })
+      .allDocs({ startkey: key, endkey: key + '\uFFFF' })
+      .then(function(response) {
+        return response.rows.length;
```

Comment:
It uses _all_docs which doesn't accept a reduce parameter. Creating a view is problematic because it's querying the user's meta db so we'd have to install and maintain a ddoc in each db. But maybe that's a necessary evil...